07_errors/jimbotech/sync_mapstore.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"math/rand"
+	"sync"
+)
+
+// SyncMapStore stores puppies threadsafe.
+type SyncMapStore struct {
+	mux sync.Mutex
+	sync.Map
```

Comment:
by embedding the map, you are making the map methods directly accessible to users of `SyncMapStore`, allowing them to bypass any checks/validation you have. You should put the `sync.Map` in as an unexported field.

Reply:
Good call.
Done